@here/harp-materials/lib/ShaderChunks/ExtrusionChunks.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export default {
+    extrusion_pars_vertex: `
+uniform float extrusionRatio;
+`,
+    extrusion_vertex: `
+transformed.z = transformed.z * extrusionRatio;
+`,
+    extrusion_pars_fragment: `
+uniform float extrusionRatio;
+`,
+    extrusion_fragment: `
+gl_FragColor.a *= smoothstep( 0.0, 0.25, extrusionRatio );
```

Comment:
Why do we need to animate the alpha as well? Why is the factor at the end of the animation 0.25?

Reply:
I decided to add alpha to make animation a bit smoothier. The appearance of extruded objects at one moment, before it starts extruding, disharmonize with the smooth animation of extrusion.
Adding alpha transition in the first quarter of animation ( 0.0, 0.25 ) makes user better experience. On my taste, I think it's better.

Reply:
I agree with Sergii, and it's also what 2gis is doing, however I'm not sure it's in the scope of the task given that it may conflict with the fading in of tiles? Or that could be kept and dealt with especially if Sergii works on this other feature.